### PR TITLE
Update jest config file

### PR DESCRIPTION
**Pull request by eliamartani**
## Description
This PR fixes one issue I found when running `npm test`.

## Motivation and Context

For testing, the `??` is not recognized by Jest, setting `babelConfig` helps running `npm test`.

## How Has This Been Tested?

I retested the script after implementing it and the issue was gone.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Author's Checklist:

- [x] My code focuses on the main motivation and avoids scope creep.
- [ ] My code passes current tests and adds new tests where possible.
- [ ] My code is [SOLID](https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)) and [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).
- [ ] I have updated the documentation as needed.

## Reviewer's Checklist:
- [x] I kept my comments to the author positive, specific, and productive.
- [x] I tested the code and didn't find any new problems.
- [x] I think the motivation is good for the project.
- [x] I think the code works to satisfies the motivation.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   },
   "dependencies": {
     "gpu.js": "^2.10.4",
-    "thaw.js": "^2.1.0"
+    "thaw": "^2.0.1"
```
Comment (mubaidr) [inline package.json]: We are using this package: 
https://www.npmjs.com/package/thaw.js

**Review by mubaidr — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (eliamartani) [reply package.json]: You're right, I messed up the plugins

**Comment by mubaidr** [comment]
This sure does fix the issue. 👍🏼 

But what if we override `tsconfig` for tests to replace with `"target": "esnext"` with `"target": "es6"`, this will make sure typescript compiles newest feature to `es6` which `jest` understands. 

So instead of `Babel -> Typescript > jest` we can do `typescript > jest`

**Comment by eliamartani** [comment]
> This sure does fix the issue. 👍🏼
> 
> But what if we override `tsconfig` for tests to replace with `"target": "esnext"` with `"target": "es6"`, this will make sure typescript compiles newest feature to `es6` which `jest` understands.
> 
> So instead of `Babel -> Typescript > jest` we can do `typescript > jest`

It works as well. `ts-jest` will receive `tsConfig`* instead with the configuration to be used, `tsconfig.test.json`

```json
{
  "tsConfig": "tsconfig.test.json"
}
```

tsconfig.test.json:
```json
{
  "compilerOptions": {
    "allowJs": true,
    "esModuleInterop": true,
    "module": "es6",
    "moduleResolution": "node",
    "target": "es6"
  }
}
```

What do you think @mubaidr ?

* I based myself on https://kulshekhar.github.io/ts-jest/user/config/

**Comment by mubaidr** [comment]
Sound good enough! 

lets use `ts-jest/presets/js-with-ts` preset and avoid `babel` for tests only by extending `tsconfig.json` to `tsconfig.test.json`

**Comment by eliamartani** [comment]
Changes made. Tested with the preset and it worked.

**Pull request merged**